### PR TITLE
Update migration guide

### PR DESCRIPTION
**Pull request by bjohansebas**
Several undocumented changes have been added in version 5, as referenced in https://github.com/expressjs/expressjs.com/issues/1479#issuecomment-2102703450:

### CODE REVIEW

**Comment by wesleytodd** [comment]
Thanks for the ping! I hadn't seen this until now. I am working on a blog post for the v5 release now, and it would be awesome if we could land this before so I can link to it. I also ping'd @danieldelcore who [did some codemod work](https://x.com/danieldelcore/status/1834391250897981781), would be good if we can team up for this.

**Comment by bjohansebas** [comment]
@wesleytodd It would be great to add codemod, it would make the transition to the new version much easier. I can help with that.

File: en/guide/migrating-5.md [hunk 1]
```diff
@@ -113,26 +129,33 @@ The `res.sendfile()` function has been replaced by a camel-cased version `res.se
...
+- The special `*` path segment behavior has been removed. This means that `*` is no longer a valid path and `/foo/*/bar` now matches a literal `'*'`. For the previous behaviour, a `(.*)` matching group should be used instead.
+- Named matching groups no longer available by position in `req.params`. For example, `/:foo(.*)` only captures as `req.params.foo` and not available as `req.params[0]`.
+- Regular expressions can now only be used in a matching group. For example, `/users/\\d+` is invalid and should be written as `/users/(\\d+))`.
+- It is now possible to use named capturing groups in paths using `RegExp`, for example `/\/(?<group>.+)/` would result in `req.params.group` being populated.
+- Custom prefix and suffix groups are now supported using `{}`, for example `/:entity{-:action}?` would match `/user` and `/user-delete`.
+- Unbalanced patterns now produce an error. For example `/test(foo` previously worked, but now the opening parenthesis `(` must be escaped to match the previous behavior: `/test\\(foo`.
+- As with parentheses, curly brackets also require escaping. That is, `/user{:id}` no longer matches `/user{42}` as before unless written as `/user\\{:id\\}`.
```
Comment (bjohansebas) [inline en/guide/migrating-5.md:139]: @blakeembrey I have brought in these changes from #2333, can you verify if everything is in order?

Comment (bjohansebas) [inline en/guide/migrating-5.md]: @blakeembrey  I have brought in these changes from #1317, can you verify if everything is in order?

**Comment by bjohansebas** [comment]
cc/: @expressjs/express-tc @expressjs/triagers

File: en/guide/migrating-5.md (hunk 1 above)
Comment (blakeembrey) [reply en/guide/migrating-5.md]: Unfortunately that's for pre-v8 (used in the betas), the v8 revision (used in stable) has these differences and removed a lot of these features: https://github.com/pillarjs/path-to-regexp?tab=readme-ov-file#express--4x.

Comment (bjohansebas) [reply en/guide/migrating-5.md]: I have already updated it, I think everything should be correct now

**Review by crandmck — APPROVED** [review]

**Comment by crandmck** [comment]
LGTM.  I added @UlisesGascon specifically as a reviewer, since this is based on his comment in. https://github.com/expressjs/expressjs.com/issues/1479#issuecomment-2102703450.

@UlisesGascon If you approve, please go ahead and merge this so it doesn't have to wait on me.

**Review by UlisesGascon — APPROVED** [review]

**Comment by UlisesGascon** [comment]
We can always make more changes in the future. I will merge it now :+1:

**Pull request merged**